src/jit/codegenarm64.cpp

```diff
@@ -2470,7 +2470,8 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                 // We unconditionally throw a divide by zero exception
                 genJumpToThrowHlpBlk(EJ_jmp, SCK_DIV_BY_ZERO);
 
-                // We still need to call genProduceReg
+                // We still need to call genConsumeOperands/genProduceReg for liveness update.
+                genConsumeOperands(treeNode->AsOp());
```

Comment:
If I understand Carol correctly, invoking genConsumeOperands() multiple times is not harmful. I slightly moved up genConsumeOperands() up in GenCodeForBinary(). For this failing case for division expansion, I added genConsumeOperands() in the top. PTAL again.